cmd/src/code_intel_upload_flags.go

```diff
@@ -484,14 +301,31 @@ func inferIndexRoot() (string, error) {
 // toolInfo value in the configured index file.
 //
 // Note: This function must not be called before codeintelUploadFlagset.Parse.
-func readIndexerNameAndVersion() (string, string, error) {
-	file, err := os.Open(codeintelUploadFlags.file)
+func readIndexerNameAndVersion(indexFile string) (string, string, error) {
+	file, err := os.Open(indexFile)
 	if err != nil {
 		return "", "", err
 	}
 	defer file.Close()
 
-	return upload.ReadIndexerNameAndVersion(file)
+	var metadata *scip.Metadata
+
+	visitor := scip.IndexVisitor{
+		VisitMetadata: func(m *scip.Metadata) {
+			metadata = m
+		},
+	}
+
+	// convert file to io.Reader
+	if err := visitor.ParseStreaming(file); err != nil {
+		return "", "", err
+	}
+
+	if metadata == nil {
+		return "", "", errors.New("index file does not contain metadata")
+	}
+
+	return metadata.ToolInfo.Name, metadata.ToolInfo.Version, nil
```

Comment:
You need an extra `nil` check for `ToolInfo` since the field has type `*scip.ToolInfo`, so `metadata.ToolInfo` could panic if that's not set.